platform/os_windows.go

```diff
@@ -189,6 +203,96 @@ func SetSdnRemoteArpMacAddress() error {
+	adapterNamewithVLANTag, err := ExecutePowershellCommand(cmd)
+	if err != nil {
+		return fmt.Errorf("error while executing powershell command to get VLAN Tag advance property of %s: %w", adapatername, err)
+	}
+
+	if adapterNamewithVLANTag != "" {
+		//adapter is version 4 and up since adapter's advance property consists of reg key : PriorityVLANTag
+		//set reg value for Priorityvlantag of adapater to 3
+		cmd = fmt.Sprintf("Set-NetAdapterAdvancedProperty -Name \"%s\" -RegistryKeyword \"%s\" -RegistryValue %d", adapatername, priorityVLANTagIdentifier, regvalueForVLANTag)
+		_, err := ExecutePowershellCommand(cmd)
+		if err != nil {
+			return fmt.Errorf("error while setting up registery value for PriorityVLANTag for adapater: %w", err)
+		}
+		//check if value is set or not
+		cmd = fmt.Sprintf(" Get-NetAdapterAdvancedProperty | Where-Object { $_.RegistryKeyword -like \"%s\" -and $_.Name -eq \"%s\" } | Select-Object -ExpandProperty RegistryValue", priorityVLANTagIdentifier, adapatername)
+		regvalue, err := ExecutePowershellCommand(cmd)
+		if err != nil {
+			return fmt.Errorf("error while checking registery value for PriorityVLANTag for adapater: %w", err)
+		}
+		if regvalue != fmt.Sprintf("%d", regvalueForVLANTag) {
+			return fmt.Errorf("failure while setting Mellanox Network Adapter: \"%s\" with \"%s\" property value to %d", adapatername, priorityVLANTagIdentifier, regvalueForVLANTag)
+		}
+		log.Printf("Successfully set Mellanox Network Adapter: \"%s\" with \"%s\" property value as %d", adapatername, priorityVLANTagIdentifier, regvalueForVLANTag)
+	} else {
+		//Adapter is version 3 or less as PriorityVLANTag was not found in advanced properties of mellanox adpater
+		log.Printf("Searching through CIM instances for Network devices with \"%s\" in the name", mellanoxSearchString)
+		cmd = fmt.Sprintf("Get-CimInstance -Namespace root/cimv2 -ClassName Win32_PNPEntity | Where-Object PNPClass -EQ \"Net\" | Where-Object { $_.Name -like \"%s\" } | Select-Object -ExpandProperty DeviceID", mellanoxSearchString)
+		deviceid, err := ExecutePowershellCommand(cmd)
+
+		if err != nil {
+			return fmt.Errorf("error while executing powershell command to get device id of %s: %w", adapatername, err)
+		}
+		if deviceid == "" {
+			log.Printf("No network device found with \"%s\" in description", mellanoxSearchString)
+			return nil
+		}
+
+		log.Printf("Devcie ID found and Getting PNP device properites for %s", deviceid)
+		cmd = fmt.Sprintf("Get-PnpDeviceProperty -InstanceId \"%s\" | Where-Object KeyName -EQ \"DEVPKEY_Device_Driver\" | Select-Object -ExpandProperty Data", deviceid)
+		registryKeySuffix, err := ExecutePowershellCommand(cmd)
+		if err != nil {
+			return fmt.Errorf("error while executing powershell command to get device id of %s: %w", adapatername, err)
```

Comment:
should this be deviceID or adapatername?

Reply:
yes device id and also message changed now.. since it was to get registry suffix for device id